onboarding/src/Components/FormsEdit/FormDescription.js

```diff
@@ -10,86 +11,103 @@ const FormDescription = ({ formId, formData }) => {
   const [description, setDescription] = useState("");
   const [saveModal, setSaveModal ] = useState(<></>);
 
-  if(location.state && !formName) {
-    location.state.formName && setFormName(location.state.formName);
-    location.state.description && setDescription(location.state.description);
-    location.state.link && setLink(location.state.link);
-  } else {
-    if(formData && !formName) {
-      formData.title && setFormName(formData.title);
-      formData.description && setDescription(formData.description);
-      formData.link && setLink(formData.link);
+  useEffect(() => {
+    if(location.state && !formName) {
+      location.state.formName && setFormName(location.state.formName);
+      location.state.description && setDescription(location.state.description);
+      location.state.link && setLink(location.state.link);
+    } else {
+      if(formData && !formName) {
+        formData.title && setFormName(formData.title);
+        formData.description && setDescription(formData.description);
+        formData.link && setLink(formData.link);
+      };
     };
+  }, [formData]);
+
+  const hideModal = () => {
+    setSaveModal(<></>);
+  };
+
+  const popUpSaveFormDetails = (message) => {
+    setSaveModal(
+      <ModalWarning
+        handleAccept={hideModal}
+        title={"Zmiana danych formularza"}
+        message={message}
+        show={true}
+        acceptText={"Ok"}
+        id={0}
+      />
+    );
   };
 
   const handleSave = (e) => {
     e.preventDefault();
-    const isSaved = savePageDetails((res) => {},
-      formId,
-      formName,
-      link,
-      description
-    ); // pack as one argument;
-    if(isSaved) {
-      setSaveModal(
-        <ModalWarning
-          handleAccept={hideModal}
-          title={"Zmiana danych formularza"}
-          message={"Zapisano zmiany"}
-          show={true}
-          acceptText={"Ok"}
-          id={0}
-        />
-      );
+    const isValid = isValidUrl(link);
+    if(isValid || !link) {
+      const isSaved = savePageDetails(
+        popUpSaveFormDetails,
+        formId,
+        formName,
+        link,
+        description
+        ); // pack as one argument;
+    } else {
+      popUpSaveFormDetails("Błąd: Wprowadzono nieprawidłowy adres url")
     }
   };
 
-  const hideModal = () => {
-    setSaveModal(<></>);
-  }
-
   return (
-    <form>
+    <form onSubmit={ handleSave }>
```

Comment:
:+1: 
so `"Ładowanie..."` could not be send to the server :slightly_smiling_face: 
But sometimes I'm thinking if data should be checked if loaded before enabling the submit.